### ISSUE TITLE
Enable Comprehensive Historical Transaction View for Community Stake Holders

### ISSUE THREAD

**Issue by HIM92** [hicommonwealth/commonwealth#6674]
## Description
Goal: users can see all their historical stake transactions associated with their profile

- User Story: As a user on Common who has minted and burned a lot of Community Stake, and I want to review my historical transactions this week. I navigate to the MY COMMUNITY STAKE page, and then hit the TRANSACTION HISTORY TAB. Here I see all my transactions broken down by the following:
 - Community - Community Stake symbol
   - Community name is written beneath the symbol
 - Address -Community Stake smart contract address
 - Action - (either Mint or Burn)
 - Stake - Number of stake minted or burned in the transaction
 - Avg. Price - the average price of minting or burning per token
 - Total Price - the total sum paid or received by minting or burning
 - Timestamp - When the transaction took place historically, dd/mm/yyyy format or time since the transaction (whichever is an easier lift)
 - The rightmost column is linked to the transaction on etherscan. There's NO effect when the user clicks the column header. 

<img width="670" alt="Screenshot 2024-02-01 at 4 39 09 PM" src="https://github.com/hicommonwealth/commonwealth/assets/153562904/cdd2b66f-f957-4637-b000-7769cf1a98ef">

Subtickets:
- [ ] #6675 
- [ ] #6676 
- [ ] #6677

## Additional context
Figma: https://www.figma.com/file/KOEVNGXStBhmX3NnYq7MxP/Community-Stake?type=design&node-id=1812-63466&mode=design&t=7EqVHrW5J7aaRrs7-0

**Comment by mzparacha** [comment]
I discussed this with @timolegros, and it seems like we would need to find a way to query the historical stake transactions in an ideal way. An option we discussed is querying the historical stake transactions from respective blockchains directly from the front end, but not sure about feasibility. 

To prevent the development of tickets blocked by this one, I will create a separate ticket to build/integrate the historical stake transaction API and move on with other tickets
